rest/route/github.go

```diff
@@ -306,6 +321,190 @@ func (gh *githubHookApi) AddIntentForPR(pr *github.PullRequest) error {
+			"event":   gh.eventType,
+			"owner":   ownerAndRepo[0],
+			"repo":    ownerAndRepo[1],
+			"tag":     tag,
+		}))
+		return errors.Wrapf(err, "error finding projects of repo '%s'", ownerAndRepo)
+	}
+	if len(projectRefs) == 0 {
+		grip.Debug(message.Fields{
+			"source":  "github hook",
+			"message": "no projects found",
+			"ref":     event.GetRef(),
+			"event":   gh.eventType,
+			"owner":   ownerAndRepo[0],
+			"repo":    ownerAndRepo[1],
+			"tag":     tag,
+		})
+		return errors.Wrapf(err, "no projects found for repo '%s'", ownerAndRepo)
+	}
+	catcher := grip.NewBasicCatcher()
+	for _, pRef := range projectRefs {
+		// if a version for this revision exists for this project, add tag
+		existingVersion, err := gh.sc.FindVersionByProjectAndRevision(pRef.Identifier, hash)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem finding version for project '%s' with revision '%s' to add tag '%s'",
+				pRef.Identifier, hash, tag.Tag))
+			continue
+		}
+		if existingVersion == nil {
+			grip.Debug(message.Fields{
+				"source":  "github hook",
+				"message": "no version to add tag to",
+				"ref":     event.GetRef(),
+				"event":   gh.eventType,
+				"project": pRef.Identifier,
+				"branch":  pRef.Branch,
+				"owner":   pRef.Owner,
+				"repo":    pRef.Repo,
+				"hash":    hash,
+				"tag":     tag,
+			})
+			continue
+		}
+
+		grip.Debug(message.Fields{
+			"source":  "github hook",
+			"message": "adding tag to version",
+			"version": existingVersion.Id,
+			"ref":     event.GetRef(),
+			"event":   gh.eventType,
+			"branch":  pRef.Branch,
+			"owner":   pRef.Owner,
+			"repo":    pRef.Repo,
+			"hash":    hash,
+			"tag":     tag,
+		})
+
+		if err = gh.sc.AddGitTagToVersion(existingVersion.Id, tag); err != nil {
+			catcher.Add(errors.Wrapf(err, "problem adding tag '%s' to version '%s''", tag.Tag, existingVersion.Id))
+			continue
+		}
+
+		// TODO: only do this if project doesn't have a separate yaml defined
+		hasAliases, err := gh.sc.HasMatchingGitTagAlias(pRef.Identifier, tag.Tag)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem finding aliases for project '%s'", pRef.Identifier))
+			continue
+		}
+		if !hasAliases {
+			continue
+		}
+
+		p, pp, err := gh.sc.LoadProjectForVersion(existingVersion, pRef.Identifier)
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem getting project for  '%s'", pRef.Identifier))
+			continue
+		}
+		info := &model.ProjectInfo{
+			Ref:                 &pRef,
+			Project:             p,
+			IntermediateProject: pp,
+		}
+		if err != nil {
+			catcher.Add(errors.Wrapf(err, "problem populating project info for tag '%s' for project '%s'", tag.Tag, pRef.Identifier))
+			continue
+
+		}
+		metadata := model.VersionMetadata{
+			Revision: model.Revision{
+				Author:          event.GetSender().GetLogin(),
+				AuthorGithubUID: int(event.GetSender().GetID()),
+				AuthorEmail:     event.GetSender().GetEmail(),
+				Revision:        hash,
+			},
+			IsAdHoc: true,
```

Comment:
Does AdhocRequester make sense for git tags? I'm wondering if it should be a different kind of requester because later we want to add git_tags_only to project configs. Alternatively this could be part of TriggerType? But I feel like that's part of something else. @john-m-liu do you have any extra context?

Reply:
No, ad hoc most closely resembles a patch. You probably want to create a new request type for this